source/shared/cpp/ObjectModel/BaseCardElement.cpp

```diff
@@ -14,11 +14,20 @@ BaseCardElement::BaseCardElement(
     m_spacing(spacing),
     m_separator(separator)
 {
+    BaseCardElement::populateKnownPropertiesSet();
 }
 
 BaseCardElement::BaseCardElement(CardElementType type) :
     m_type(type), m_spacing(Spacing::Default)
 {
+    BaseCardElement::populateKnownPropertiesSet();
+}
+
+void BaseCardElement::populateKnownPropertiesSet(void)
+{
+    m_knownProperties.insert(AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Type));
+    m_knownProperties.insert(AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Spacing));
+    m_knownProperties.insert(AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Separator));
```

Comment:
You need all the subclasses to do this as well, or else all element properties other than these three will appear in the unknown properties json

Reply:
That's correct, I wanted to  discuss it with you.
I just simply copied over what was there in ActionCardElement.  I will make it as virtual, and have others ElementsCard override it. Do you think we should make it pure virtual?

---
In reply to: [171353942](https://github.com/Microsoft/AdaptiveCards/pull/1179#discussion_r171353942) [](ancestors = 171353942)

Reply:
So, what we do for actions is that each constructor does it's own thing. A pure virtual method that all the subclasses need to implement would probably make us less likely to miss this when we add new types, so that sounds good to me. That said, keep these shared properties somewhere in the base class so each subclass doesn't have to add them itself.

Also, could you update Actions to follow the same model so that they're consistent?  Thanks!

---
In reply to: [171363228](https://github.com/Microsoft/AdaptiveCards/pull/1179#discussion_r171363228) [](ancestors = 171363228,171353942)

Reply:
Sound good to me

---
In reply to: [171368255](https://github.com/Microsoft/AdaptiveCards/pull/1179#discussion_r171368255) [](ancestors = 171368255,171363228,171353942)